src/Client/ClientBaseHelpers.cpp

```diff
@@ -314,13 +354,15 @@ void highlight(const String & query, std::vector<replxx::Replxx::Color> & colors
                     continue;
                 }
 
+                /// TODO: @bharatnc replace the usages of countCodePointsConsistently() below with UTF8::countCodePoints()
+                /// once it's fixed to handle invalid utf-8 sequences consistently with seqLength.
                 /// Highlight the closing round bracket.
-                auto highlight_pos = UTF8::countCodePoints(reinterpret_cast<const UInt8 *>(begin), highlight_token_iterator->begin - begin);
+                auto highlight_pos = countCodePointsConsistently(query, highlight_token_iterator->begin);
                 if (highlight_pos < colors.size())
                     colors[highlight_pos] = color_stack.back();
 
                 /// Highlight the matching opening round bracket.
-                auto matching_brace_pos = UTF8::countCodePoints(reinterpret_cast<const UInt8 *>(begin), brace_stack.back().begin - begin);
+                auto matching_brace_pos = countCodePointsConsistently(query, brace_stack.back().begin);
```

Comment:
Do we also need to adjust this?

https://github.com/ClickHouse/ClickHouse/blob/73e73226799f4044530c326721e4c881501c4caa/src/Client/ClientBaseHelpers.cpp#L378-L390

Reply:
Thank you, good catch, fixed.